src/component/Calculator.js

```diff
@@ -1,43 +1,59 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = (e) => {
+    const result = calculate(this.state, e.target.textContent);
+    this.setState(result);
+  };
+
   render() {
+    const { total, next } = this.state;
     return (
       <div className="calculator">
-        <div className="calculator-display">0</div>
+        <div className="calculator-display">{ next || total }</div>
         <div className="calculator-keypad">
           <div className="input-keys">
             <div className="function-keys">
-              <button type="button" className="calculator-key key-clear">AC</button>
-              <button type="button" className="calculator-key key-sign">±</button>
-              <button type="button" className="calculator-key key-percent">%</button>
+              <button onClick={this.handleClick} type="button" className="calculator-key key-clear">AC</button>
+              <button onClick={this.handleClick} type="button" className="calculator-key key-sign">±</button>
```

Comment:
- [ ] This button does not work at all. The operation should `+/-` based on the `calcuate.js` file configuration